### PR TITLE
Issue 470 export resources bug

### PR DESCRIPTION
**Pull request by jeffnm**
This should fix the #470 issue for CORAL installations that were upgraded from 1.x versions in the past. 

During the 3.0.1 upgrade an SQL procedure will run to check if the columns in question are present or missing. If they are missing the columns will be added. 

The most significant change with this PR is that the installer's `db_tools.php` will now use `mysqli_multi_query` to run the entire SQL file, rather than attempting to split it into individual SQL statements with PHP's explode function. 

This PR also includes the first skeleton install helpers for 3.0.1

### Part 1
1. Start with a CORAL database and installation that is missing the `priceTaxExcluded`, `priceTaxIncluded`, and `taxRate` columns on the ResourcePayment table. This should be the state of any CORAL installation that was upgraded from 1.x to 2.0 or 3.0. Fresh installations to 2.0 or 3.0 will have these columns.
2. Attempt to download the Excel file export per #470. It will fail with a server error (blank screen).
3. Apply this pull request and run through the CORAL upgrade. 
4. Attempt step 2 again. The Excel file should download with the Resources search results you were displaying.

### Part 2
Complete a fresh installation of CORAL using this pull request to verify that there are no problems with the new `db_tools.php`

### CODE REVIEW

**Comment by jeffnm** [comment]
I'm labeling this release blocking for 3.0.1. This PR needs to be included for 3.0.1 since it contains installer updates as well. 

If someone else could look at it and verify one or both of the install/update cases, I'd appreciate it.

**Comment by LibClaire** [comment]
Hi, I tried to test this but was unsuccessful. Maybe it won't work the way I tested it which was by deleting the columns from my 3.0.0 database. 
After the columns were deleted I could reproduce the error with the export.
I applied the patch and the installer appeared asking me to grant and revoke permissions for 2 configuration.ini files and was then done. However the columns have not been added to the database and I still cannot export resources. I get the same php error as before.
LIke I say, perhaps I cannot test it this way. I checked and all commits were successfully applied. My Coral version is now '3.0.1' and the footer says development.

**Comment by poacherone** [comment]
I tried part 1 of this PR over the weekend. I cheated a little an simply deleted the columns from my table in my 3.0.0 install, applied the PR and ran the Coral upgrade. It looks like it runs through the upgrade (I'm asked for db credentials etc.) but the columns aren't added to the database. Is this because I didn't do an installation of 1.x and then upgraded? Does the installer check versions? 

Or is it an issue with the SQL query? (I also tried running it directly, but that didn't work either.)

**Comment by poacherone** [comment]
So basically, I experienced the same as @LibClaire reported above. ^^

Also have version and footer changed.

**Comment by jeffnm** [comment]
Thanks @LibClaire and @poacherone let me look into it. It's supposed to check if those columns are present and, if they aren't, add them, which did work on my test server, but it might be something specific to my version or my database. 

It sounds like the installer processed, but the SQL file didn't get applied for some reason.

**Comment by jeffnm** [comment]
It looks like it's a collation mismatch error. I think I can fix that, but this is revealing that the error handling for the mysqli_multi_query is not working properly.

**Comment by jeffnm** [comment]
I pushed a commit that has better error handling for the mysql error messages so we should actually catch the error message, as well as making the parameter explicitly utf8 which solved the mismatch for my testing.

**Comment by jeffnm** [comment]
@LibClaire and @poacherone would you be able to test this again? I think making the collation explicitly utf8 should have fixed it, but I definitely would like some more tests since it didn't work for you two previously.

**Comment by poacherone** [comment]
Tested again and the installer now runs properly, and the database is updated with the missing columns. The export however, still doesn't work (blank page in Firefox, http 500 error in Chrome).

**Comment by LibClaire** [comment]
Seems to be working. First attempt - worked straight away so I wanted to try a second test to be sure. 

I deleted the columns from ResourcePayment - I reapplied the patch but it didn't work, it seems I had to also change the version in common/configuration.ini back to 3.0.0 as it was still showing as development. 
Once I did that the installer ran again as expected, the columns were added to the table, version is set to 3.0.1 and I can download the file.

Looks good to me

**Comment by poacherone** [comment]
After @LibClaire's comment, I went back and looked more closely at the logs. The export failed because there was missing columns in the Resource-table (currentStartDate and currentEndDate). The were missing because of testing [Pr 510](https://github.com/coral-erm/coral/pull/510), which previously dropped those tables if I recall correctly. Since this has changed, and the columns aren't dropped, it shouldn't affect this PR.

So, works as intended!

**Comment by jeffnm** [comment]
Thank you both, @LibClaire and @poacherone. It's much appreciated!

I'm concerned that the export requires the Resource table currentStartDate and currentEndDate columns, when those are considered to be deprecated now, but I guess that's a different issue. 

@veggiematts can you merge this based on those two tests? I don't want to merge my own PR, especially when it is a little more involved like this one.

File: templates/footer.php [hunk 1]
```diff
@@ -24,7 +24,7 @@
 <div class="push">&nbsp;</div>
 </div>
 
-<div class="footer"><?php echo _("Copyright");?> &copy; 2018. <?php echo _("CORAL version");?> 3.0.0<br/>
+<div class="footer"><?php echo _("Copyright");?> &copy; 2018. <?php echo _("CORAL version");?> Development Branch<br/>
```
Comment (veggiematts) [inline templates/footer.php]: Is this change intended?
